force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -139,40 +224,39 @@ public with sharing class FlowControlLoopsCollections {
      */
     public static String returnHelloWorld() {
         // Initialize the result String
+        String resultString = 'Hello World!; ';
+        //use for loop to append 'Hello World!'to the result string three times
+        for (Integer x = 0; x < 2; x++){
```

Comment:
This should have been < 3 and make your starting string empty. Clearer to see how many loops.